dhis-2/dhis-services/dhis-service-core/pom.xml

```diff
@@ -260,7 +259,7 @@
     </dependency>
     <dependency>
       <groupId>org.junit.jupiter</groupId>
-      <artifactId>junit-jupiter</artifactId>
```

Comment:
My gut feeling is that we should not change this to junit-jupiter-api. https://junit.org/junit5/docs/current/user-guide/#running-tests-build-maven-engines-configure also states dependency on artifact `junit-jupiter`. It says

> To configure support for JUnit Jupiter based tests, configure test scoped dependencies on the JUnit Jupiter API and the JUnit Jupiter TestEngine implementation similar to the following.

AFAIK API is for the things like the `@Test` and assertions, the engine is then the thing running the tests. I suspect that if you only have `junit-jupiter-api` left in DHIS2, tests will not run. See https://maven.apache.org/surefire/maven-surefire-plugin/usage.html `Using Different Testing Providers` Surefire knows we use JUnit 5 also because the `junit-jupiter-engine` dependency is in our project.

I think we should tell maven-dependency-plugin to ignore it being unused instead https://github.com/dhis2/dhis2-core/blob/7187c6e2c5c6089ef6c49c90ae2cdfdbf09f3523/dhis-2/pom.xml#L712

Reply:
I agree